src/module.c

```diff
@@ -6704,6 +6721,50 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+/* Check if the command can be executed by the user created through the redis module
+ * interface, according to the ACLs associated to it.
+ *
+ * If the user can execute the command, REDISMODULE_OK is returned, otherwise
+ * REDISMODULE_ERR is returned. */
+int RM_ACLCheckCommandPerm(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    int keyidxptr;
+    struct redisCommand *cmd;
+
+    /* Find command */
+    if ((cmd = lookupCommand(argv[0]->ptr)) == NULL) {
+        errno = ENOENT;
+        return REDISMODULE_ERR;
+    }
+
+    if (ACLCheckAllUserCommandPerm(ctx->client->user, cmd, argv, argc, &keyidxptr) != ACL_OK) {
```

Comment:
What if it is used from a ThreadSafeCtx that does not attach to any block client? Will it crash here? Also, if its a ThreadSafeCtx that that was attached to this user but until it was created by the background thread the user was already dropped?

Reply:
I guess this question is relevant also to the rest of this API

Reply:
maybe we need to add a variant of the API that takes a user name as input rather than rely on the connection, and obviously and API to get the current user.
if the user is already disconnected or modified by the time the module tries to use it, i guess the module should fail (get denied)

Reply:
I was thinking maybe it might be a good idea to introduce a credentials object. All those new functions can get a credentials object (instead of RedisModuleCtx*) and we can also add an API to get credentials object from RedisModuleCtx. The credentials object can leave even after the user has changed/deleted and can be used on another thread.

Reply:
i'm not sure this is the desired path (at least at this point in time).
seems simpler and more in-line with other things to let the module pass the user name around, and if it was modified or deleted, the validation should just get denied.
@yossigo WDYT?

Reply:
@oranagra I agree with you, passing the user name is better, this way if the user permission change it will still work correctly.

Reply:
@MeirShpilraien @oranagra IIUC this is what @yossigo already mentioned no?
"Also, I am not sure about the coupling with RedisModuleCtx here. Doesn't it make more sense to take a RedisModuleUser here, and have a separate function that can extract RedisModuleUser from the context?"

Reply:
@YaacovHazan yep missed that ... but I do believe the username suggested by @oranagra is a good idea...

Reply:
@MeirShpilraien @oranagra Why pass usernames around when we can use `RedisModuleUser`?

Reply:
@yossigo from my perspective the user name is an opaque struct that represents the user, so I think it's fine if it's `RedisModuleUser` as long as we handle permission changes/deletion and we make sure `RedisModuleUser` can continue living even after the actual user got deleted.

Reply:
and of course, if the user was deleted and the module continues using its `RedisModuleUser` then we get permission deny on everything as @oranagra mentioned.

Reply:
what's worse, if the module keeps a `RedisModuleUser` for later use, and the user is deleted, then the `->user` member here points to released memory!

Reply:
This makes sense, I'd still feel more comfortable having some opaque `RedisModuleUserId` and not commit to the fact that it's a string / the username.